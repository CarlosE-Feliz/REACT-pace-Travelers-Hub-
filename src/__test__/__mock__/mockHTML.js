/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable import/prefer-default-export */
import { useSelector, useDispatch } from 'react-redux';
import { rocketReseve } from '../../redux/rocketsAPI';

export const Data = (props) => {
  const post = useSelector((state) => state.rock);
  const dispatch = useDispatch();
  const reserved = () => {
    dispatch(rocketReseve(post, props.id));
  };

  return (
    <div key={props.id}>
      <img src={props.img} alt="img" />
      <div>
        <h1>
          {props.name}
        </h1>
        <p>
          {props.reserved ? <span className="badge bg-success">Reserved</span> : null}
        </p>
        {props.reserved ? <button type="button" onClick={reserved} id={props.id} className="btn btn-secondary">Cancel Reservation</button> : <button type="button" onClick={reserved} id={props.id} className="btn btn-primary">Reserve Rocket</button>}
      </div>
    </div>
  );
};
