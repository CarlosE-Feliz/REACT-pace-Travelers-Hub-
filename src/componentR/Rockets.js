/* eslint-disable no-console */
// import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
// import { bindActionCreators } from 'redux';
import { rocketReseve } from '../redux/API';

import {
  Div, Section, Picture, Info,
} from './styles';

const Rockets = () => {
  const post = useSelector((state) => state.rock);
  const dispatch = useDispatch();
  const reserved = (e) => {
    dispatch(rocketReseve(post, e.target.id));
  };
  return (
    <Div>
      {post.map((post) => (
        <Section key={post.id}>
          <Picture src={post.img} alt="img" />
          <Info>
            <h1>
              {post.name}
            </h1>
            <p>
              {post.reserved ? <span style={{ margin: '5px' }} className="badge bg-success">Reserved</span> : null}
              {post.desc}
            </p>
            {post.reserved ? <button type="button" onClick={reserved} id={post.id} className="btn btn-secondary">Cancel Reservation</button> : <button type="button" onClick={reserved} id={post.id} className="btn btn-primary">Reserve Rocket</button>}
          </Info>
        </Section>
      ))}
    </Div>
  );
};

export default Rockets;
