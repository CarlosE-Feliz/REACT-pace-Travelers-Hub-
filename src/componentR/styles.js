/* eslint-disable import/prefer-default-export */
import styled from 'styled-components';

export const Div = styled.div`
    margin: 1rem;
`;

export const Section = styled.div`
    display: flex;
    margin: 1rem;
`;

export const Picture = styled.img`
    width: 250px;
    margin: 2rem;
`;

export const Info = styled.div`
    margin: 2rem;
`;
