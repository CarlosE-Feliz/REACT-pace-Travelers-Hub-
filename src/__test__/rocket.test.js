/* eslint-disable no-unused-vars */
/* eslint-disable no-console */
import { render, fireEvent, screen } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { Rockets } from '../componentR/styles';
import store from '../redux/config';
import { Data } from './__mock__/mockHTML';

describe('Test button Rocketss', () => {
  it('snapshot test', async () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Rockets>
            <Data key="0" id="0" name="test" type="Test test" image="test" reserved />
          </Rockets>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Reserved badge in Rockets component', () => {
    render(
      <Provider store={store}>
        <Data key="0" id="0" name="test" type="Test test" image="test" reserved />
      </Provider>,
    );
    screen.getByText('Cancel Reservation');
    expect(screen.getByText('Reserved')).toBeInTheDocument();
  });

  test('Cancel Rockets button', () => {
    render(
      <Provider store={store}>
        <Data key="0" id="0" name="test" type="Test test" image="test" reserved />
      </Provider>,
    );
    screen.getByText('test');
    expect(screen.getByText('Cancel Reservation')).toBeInTheDocument();
  });

  test('Reserve Rockets button', () => {
    render(
      <Provider store={store}>
        <Data key="0" id="0" name="test" type="Test test" image="test" reserved />
      </Provider>,
    );
    screen.getByText('test');
    fireEvent.click(screen.getByRole('button'));
    screen.debug();
  });
});
