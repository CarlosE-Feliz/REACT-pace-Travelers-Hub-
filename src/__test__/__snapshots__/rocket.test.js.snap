// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test button Rocketss snapshot test 1`] = `
<div
  className="sc-jSgvzq QWMvE"
>
  <div>
    <img
      alt="img"
    />
    <div>
      <h1>
        test
      </h1>
      <p>
        <span
          className="badge bg-success"
        >
          Reserved
        </span>
      </p>
      <button
        className="btn btn-secondary"
        id="0"
        onClick={[Function]}
        type="button"
      >
        Cancel Reservation
      </button>
    </div>
  </div>
</div>
`;
